ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/SystemAuthenticationFunctionalTest.java

```diff
@@ -226,25 +226,19 @@ public void shouldHeartbeatSuccessfully() throws InterruptedException {
       Thread.sleep(6000);
 
       // When:
-      final HostStatusEntity host0Up = waitForRemoteServerToChangeStatus(
-          REST_APP_0, host0, HighAvailabilityTestUtil::remoteServerIsUp, Optional.of(USER1))
-          .getClusterStatus().get(host0);
-      final HostStatusEntity host1Up = waitForRemoteServerToChangeStatus(
-          REST_APP_0, host1, HighAvailabilityTestUtil::remoteServerIsUp, Optional.of(USER1))
-          .getClusterStatus().get(host1);
-      ClusterStatusResponse response = sendClusterStatusRequest(REST_APP_0, Optional.of(USER1));
+      // wait for a single response that shows both are alive, this is necessary
+      // because if we wait for them separately the test can be flaky as one might
+      // go down while we wait for the other
+      final ClusterStatusResponse response = waitForRemoteServerToChangeStatus(
+          REST_APP_0,
+          clusterStatus -> clusterStatus.get(host0).getHostAlive()
+              && clusterStatus.get(host1).getHostAlive(),
+          Optional.of(USER1)
+      );
 
       // Then:
-      final String format = "Expected %s to be up after confirming response %s, "
-          + "but latest response %s diverged";
-      assertThat(
-          String.format(format, host0, host0Up, response.getClusterStatus().get(host0)),
-          response.getClusterStatus().get(host0).getHostAlive(),
-          is(true));
-      assertThat(
-          String.format(format, host1, host1Up, response.getClusterStatus().get(host1)),
-          response.getClusterStatus().get(host1).getHostAlive(),
-          is(true));
+      assertThat( response.getClusterStatus().get(host0).getHostAlive(), is(true));
```

Comment:
yes, I realize that this is basically testing exactly the condition we're waiting for - but it felt weird without a "then" portion